src/test/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilterUnitTest.java

```diff
@@ -0,0 +1,571 @@
+            final EvidenceFeatures[] matrix = new EvidenceFeatures[numRows];
+            if (numRows == 0) {
+                return matrix;
+            }
+            matrix[0] = new EvidenceFeatures(getDoubleArrayFromJsonArrayNode(dataNode.get(0)));
+            final int numColumns = matrix[0].length();
+            for (int row = 1; row < numRows; ++row) {
+                matrix[row] = new EvidenceFeatures(getDoubleArrayFromJsonArrayNode(dataNode.get(row)));
+                final int numRowColumns = matrix[row].length();
+                Utils.validateArg(numRowColumns == numColumns, "Rows in JSONArray have different lengths.");
+            }
+            return matrix;
+        }
+
+        private static EvidenceFeatures[] getFVecArrayFromPandasJsonNode(final JsonNode dataNode) {
+            if(!dataNode.isObject()) {
+                throw new IllegalArgumentException("dataNode does not encode a valid pandas DataFrame");
+            }
+            final int numColumns = dataNode.size();
+            if(numColumns == 0) {
+                return new EvidenceFeatures[0];
+            }
+
+            final String firstColumnName = dataNode.fieldNames().next();
+            final int numRows = getColumnArrayNode(dataNode.get(firstColumnName)).size();
+            final EvidenceFeatures[] matrix = new EvidenceFeatures[numRows];
+            if (numRows == 0) {
+                return matrix;
+            }
+            // allocate each EvidenceFeatures in matrix
+            for(int rowIndex = 0; rowIndex < numRows; ++rowIndex) {
+                matrix[rowIndex] = new EvidenceFeatures(numColumns);
+            }
+            int columnIndex = 0;
+            for(final Iterator<Map.Entry<String, JsonNode>> fieldIter = dataNode.fields(); fieldIter.hasNext();) {
+                // loop over columns
+                final Map.Entry<String, JsonNode> columnEntry = fieldIter.next();
+                final JsonNode columnArrayNode = getColumnArrayNode(columnEntry.getValue());
+                Utils.validateArg(columnArrayNode.size() == numRows,
+                        "field " + columnEntry.getKey() + " has " + columnArrayNode.size() + " rows (expected " + numRows + ")");
+                // for each FVec in matrix, assign feature from this column
+                int rowIndex = 0;
+                for(final JsonNode valueNode: columnArrayNode) {
+                    final EvidenceFeatures fVec = matrix[rowIndex];
+                    fVec.setValue(columnIndex, valueNode.asDouble());
+                    ++rowIndex;
+                }
+                ++columnIndex;
+            }
+            return matrix;
+        }
+
+        private static JsonNode getColumnArrayNode(final JsonNode columnNode) {
+            return columnNode.has(VALUES_NODE) ? columnNode.get(VALUES_NODE) : columnNode.get(CODES_NODE);
+        }
+
+        static double[] getDoubleArrayFromJsonNode(final JsonNode vectorNode) {
```

Comment:
These parsing methods are long and tedious (for you to write and others to read). I'm wondering if there are libraries you could use to ease this parsing and cut down on the lines of code. What about GSON, which we include in our dependenies? See here: https://gist.github.com/leoleozhu/5964515 . Could we use that instead to do this parsing work? It's not a huge deal since this is test code but it'd be nice to cut down some of this boilerplate.

Reply:
I picked jackson quasi-randomly (it was among the fastest json libraries that was well-liked). The sample GSON code you showed looked simple to read in an array, but the data I'm sending over is nested (multiple objects, tables, matrices, arrays). My recollection is that GSON requires object deserialization which is complex if done directly, but has helper tools. I'd have to study up (or be given instruction) on the best way to do that.
On the other hand, if I get a java training tool running, we can just serialize / deserialize and cut out quite a bit of code.

Reply:
OK, maybe you want to put in an issue or at least add a TODO comment reminding us to see if that parsing stuff can be simplified.

Reply:
Ok. I'm thinking create an issue along the lines of Create Java training tool for XGBoostEvidenceFilter, and have "Eliminate or simplify JSON parsing for test data" as a requirement in the description of that issue. Does that make sense?